### ISSUE TITLE
[no-unnecessary-qualifier] false positive with module augmentation in a .ts module

### ISSUE THREAD

**Issue by ark120202** [typescript-eslint/typescript-eslint#1993]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unnecessary-qualifier": ["error"]
  }
}
```

```ts
import * as ts from "typescript";

declare module "typescript" {
    function createDiagnosticReporter(system: ts.System): any;
}
```

**Expected Result**

No errors, since removing `ts.` makes this code invalid:

```
TS4078: Parameter 'system' of exported function has or is using private name 'System'.
```

**Actual Result**

```
Qualifier is unnecessary since 'System' is in scope
```

**Additional Info**

This behavior is different for .d.ts files - removing `ts.` in declaration files is valid.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.31.0` |
| `@typescript-eslint/parser`        | `2.31.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `14.2.0` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
I cannot repro this:
![image](https://user-images.githubusercontent.com/7462525/81483912-6161c080-91f6-11ea-97bd-535e33f56650.png)

Could you please provide a repro repo?

**Comment by ark120202** [comment]
It is the issue specifically for `.ts` files, in `.d.ts` files it works as expected, so just rename your repro file to `test.ts`

**Comment by bradzacher** [comment]
why are you augmenting a module declaration from a non-declaration file?

**Comment by ark120202** [comment]
Yes, pretty much. I guess that might look weird in this case, but it works in TypeScript, so I think the rule should follow it's behavior. It's also shown in the  example on https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation page, for cases where you want to augment the module and add the implementation for this augmentation

**Comment by Vinccool96** [comment]
It looks to be resolved, if someone could confirm
